include/envoy/network/dns.h

```diff
@@ -62,6 +62,17 @@ class DnsResolver {
    */
   virtual ActiveDnsQuery* resolve(const std::string& dns_name, DnsLookupFamily dns_lookup_family,
                                   ResolveCb callback) PURE;
+
+  /**
+   * Initiate an async DNS resolution for an SRV record.
+   * @param dns_name supplies the DNS name to lookup.
+   * @param dns_lookup_family the DNS IP version lookup policy.
+   * @param callback supplies the callback to invoke when the resolution is complete.
+   * @return if non-null, a handle that can be used to cancel the resolution.
+   *         This is only valid until the invocation of callback or ~DnsResolver().
+   */
+  virtual ActiveDnsQuery* resolveSrv(const std::string& dns_name, DnsLookupFamily dns_lookup_family,
```

Comment:
I know the existing API does it this way, but I'd be interested if we could make ActiveDnsQuery RAII.

Reply:
Would you be willing to explain what you mean here more?  The ActiveDnsQuery abstract class only has one method and no data members.  The PendingResolution struct is derived from ActiveDnsQuery and contains data, is that what you want to be RAII?  Do you know why a struct was used instead of a class? Was it just to avoid another class definition?

Reply:
I think I was suggesting that the returned `ActiveDnsQuery` be a unique ptr. The idea is that if this returned object is then destructed, it gives you automagic cancellation of the request.

Reply:
@htuch Are you saying it's better to return a `std::unique_ptr<ActiveDnsQuery>`  instead of `ActiveDnsQuery*`? I'm not quite sure how it actually cancels the pending dns request. Mind elaborating a bit more? (Sorry if the answer is obvious as I'm pretty new to C++ :) )

Reply:
Right, return `std::unique_ptr<ActiveDnsQuery>`. The destructor for `ActiveDnsQuery` can then perform the cancellation.